src/formatDate.js

```diff
@@ -8,7 +8,24 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = Array.from(fromFormat).pop();
+  const newFormat = Array.from(toFormat);
+  const newSeparator = newFormat.pop();
+  const dataObject = {};
+
+  for (const value of date.split(oldSeparator)) {
+    const key = fromFormat.shift();
+
+    dataObject[key] = value;
+
+    if (key === 'YY') {
+      dataObject.YYYY = value < 30 ? '20' + value : '19' + value;
+    } else if (key === 'YYYY') {
+      dataObject.YY = value.substring(2);
+    }
+  }
+
+  return newFormat.map((call) => dataObject[call]).join(newSeparator);
```

Comment:
The variable name `call` does not accurately describe its contents. A better name would be `formatPart` or `datePart` to reflect that it represents a part of the `toFormat` array.